packages/ra-core/src/dataProvider/useGetOne.ts

```diff
@@ -64,24 +68,22 @@ export const useGetOne = <RecordType extends RaRecord = any, ErrorType = Error>(
     const onErrorEvent = useEvent(onError);
     const onSettledEvent = useEvent(onSettled);
 
-    const result = useQuery<RecordType, ErrorType>({
+    const result = useQuery<GetOneResult<RecordType>, ErrorType>({
```

Comment:
unfortunately, this is a breaking change: any other hook writing into the `getOne` cache (e.g., `useUpdate`) would need to be updated...